### PR TITLE
Upgrade rkt to 1.29.0

### LINKED ISSUE #91: Upgrade rkt to 1.29.0
Upstream: [v1.29.0](https://github.com/rkt/rkt/releases/tag/v1.29.0)

Upstream update checklist:
- [x] Ensure upstream version is latest.
- [x] Review changelogs for breaking changes.
- [x] Verify upstream release hash.
- [x] Add `upgrade-rkt-1.29.0` branch with update.
- [x] Implement any repackaging changes.
- [x] Test cluster deployment with changes.
- [ ] Merge changes.

### PR DESCRIPTION
**Pull request by celskeggs**
Fixes #91.

Note that the repacked coreos archive is, against homeworld policy, an archive of upstream binaries. Fixing that is tracked at #92.

These changes have been tested.

### CODE REVIEW

File: building/upstream.log [hunk 1]
```diff
@@ -34,3 +34,14 @@
...
+- output: coreos_restructured-1478.0.0.cpio.gz
+  last: 2018-01-30
+
+  source: https://alpha.release.core-os.net/amd64-usr/1478.0.0/coreos_production_pxe_image.cpio.gz
+  signature: https://alpha.release.core-os.net/amd64-usr/1478.0.0/coreos_production_pxe_image.cpio.gz.sig
+
+  signing-key: 04127D0BFABEC8871FFB2CCE50E0885593D2DCB4
+  signing-key-retrieved-from: https://coreos.com/security/image-signing-key/
+
+  process: restrip-coreos.sh
```
Comment (ichung) [inline building/upstream.log]: Could you add an entry for `rkt-1.29.0.tar.xz`?

Comment (ichung) [inline building/upstream.log]: I'm not sure exactly how the `restrip-coreos.sh` script expects to receive its input.  Maybe make the instructions more explicit?

Comment (celskeggs) [reply building/upstream.log]: Added.

Comment (celskeggs) [reply building/upstream.log]: I've added explanatory checks to the script. Does that help?

**Comment by celskeggs** [comment]
@ichung, @krawthekrow: re-review?

File: building/upstream.log [hunk 2]
```diff
@@ -34,3 +34,24 @@
   comment: mostly matches manual git-archive export, except for a couple of minor versioning differences, as expected
 
   compress: gunzip, xz
+
+- output: rkt-1.29.0.tar.xz
+  last: 2018-01-31
+
+  git: https://github.com/rkt/rkt
+  tag: v1.29.0
+  compress: xz
```
Comment (ichung) [inline building/upstream.log]: I managed to reproduce this one, but a line saying `git archive v1.29.0 --prefix='rkt-1.29.0/'` would have been a useful hint.

File: building/upstream.log [hunk 3]
```diff
@@ -34,3 +34,24 @@
...
+
+  signing-key: 18AD5014C99EF7E3BA5F6CE950BDD3E0FC8A365E
+  signing-key-retrieved-from: https://coreos.com/security/app-signing-key/
+
+- output: coreos_restructured-1478.0.0.cpio.gz
```
Comment (ichung) [inline building/upstream.log]: not reproducible.

**Comment by celskeggs** [comment]
@ichung: I improved the reproducibility of the tarball as per previous discussion. Thoughts on this solution, while I work out the last reproducibility issues?

**Comment by ichung** [comment]
looks good to me

**Comment by celskeggs** [comment]
@ichung: Should be fully reproducible now.

**Comment by celskeggs** [comment]
@ichung or @krawthekrow: will either of you have time to review this PR soon? I have a decent bit more work that's waiting on this PR.

**Comment by celskeggs** [comment]
I've verified reproduction with a separate system. I'm bypassing further review on this PR, as per the conversation on zephyr. (Also because there's very little changed from a version that passed review.)

**Pull request merged**